engine/src/inputs/controller.cpp

```diff
@@ -0,0 +1,99 @@
+// Copyright (c) 2023-present Genesis Engine contributors (see LICENSE.txt)
+
+#include "inputs/controller.hpp"
+
+#include <vector>
+#include <map>
+#include <string>
+#include <iostream>
+
+namespace gen
+{
+	namespace controllerManager
+	{
+		namespace
+		{
+			std::vector<Controller> g_connectedControllers; //Max 16 (because of GLFW)
+		}
+
+		void init()
+		{
+			glfwSetJoystickCallback(gamePadCallback);
+
+			int alreadyConnectedGamePadCount = 0;
+
+			while (glfwJoystickPresent(alreadyConnectedGamePadCount) != 0)
+			{
+				if (glfwJoystickIsGamepad(alreadyConnectedGamePadCount))
+					g_connectedControllers.emplace_back(alreadyConnectedGamePadCount);
+				alreadyConnectedGamePadCount++;
+
+				if (alreadyConnectedGamePadCount == GLFW_JOYSTICK_LAST)
+					break;
+			}
+		}
+
+		void updateControllers()
+		{
+			for (auto & connectedController : g_connectedControllers)
+				if (connectedController.isActive())
+					connectedController.update();
```

Comment:
Up to @Rinzii but I'd recommend using braces all the time; this risks bugs when more lines are added, especially in merge conflicts.

Reply:
I agree with preferring braces always as it is less error prone and generally far more concise in my opinion.